ragstack/colbert/cassandra_retriever.py

```diff
@@ -110,71 +110,68 @@ def retrieve(
         top_k = max(math.floor(len(query_encodings) / 2), 16)
         logging.debug(f"query length {len(query)} embeddings top_k: {top_k}")
 
-        # find the most relevant documents
-        docparts: Set[Tuple[Any, Any]] = set()
-        doc_futures = []
+        # find the most relevant chunks
+        chunks: Set[Tuple[str, int]] = set()
+        ann_futures: List[ResponseFuture] = []
         for qv in query_encodings:
             # per token based retrieval
-            doc_future = self.vector_store.session.execute_async(
+            future = self.vector_store.session.execute_async(
                 self.vector_store.query_colbert_ann_stmt, [list(qv), top_k]
             )
-            doc_futures.append(doc_future)
+            ann_futures.append(future)
 
-        for future in doc_futures:
+        for future in ann_futures:
             embeddings = future.result()
-            docparts.update(
+            chunks.update(
                 (embedding.doc_id, embedding.chunk_id) for embedding in embeddings
             )
 
         # score each document
-        scores = {}
-        futures = []
-        for doc_id, chunk_id in docparts:
+        chunk_scores: Dict[Tuple[str, int], Tensor] = {}
+        score_futures: List[Tuple[ResponseFuture, str, int]] = []
+        for doc_id, chunk_id in chunks:
             future = self.vector_store.session.execute_async(
                 self.vector_store.query_colbert_chunks_stmt, [doc_id, chunk_id]
             )
-            futures.append((future, doc_id, chunk_id))
+            score_futures.append((future, doc_id, chunk_id))
 
-        for doc_id, chunk_id in docparts:
-            # blocking call until the future is done
+        for future, doc_id, chunk_id in score_futures:
```

Comment:
this was the main issue... we were only waiting for one future from the above to complete... not all of them.

everything else is just renaming and type-hinting to make the code easier to read.